### DISCUSSION TITLE
Feedbacks

### DISCUSSION THREAD

**Discussion by Jaysmito101** [Jaysmito101/TerraForge3D#6] [General]
Here all are welcome to give any feedbacks for this!

**Comment by razvanab** [comment]
An erosion generator will be nice to have.

**Reply by Jaysmito101** [reply]
Hi, I have plans to do that. But for now to get a similar effect you could 

Firstly if you just want some fine details look you could add a noise layer set type to random and have a very small strength  the result might be good enough . 

Secondly layering noise layers is not the only workflow you can use the node editor based workflow instead in which you could do a lot more . also you could write up a script node in lua to implement custom logic and do anything you want. 

Lastly yeah i am planning to add hydraulic erosion but honestly i don't know how that works so it will take some for me to learn and then implement. But i will try to do it very soon!

**Reply by razvanab** [reply]
I'll try that.
Thanks for the replay.

**Reply by Jaysmito101** [reply]
Just a small requests,as this projects is quite new I would be really grateful if you could post the screenshot of your creations to https://github.com/Jaysmito101/TerraGen3D/issues/10 if you do not mind.

**Reply by Jaysmito101** [reply]
@razvanab I have an update. Added Basic  Erosion https://github.com/Jaysmito101/TerraGen3D/commit/1f72c10e0c7973edb58067ceaa69e0b976309093.

Something about it:

* I could not add just add a button to start erosion anywhere!
* I added a completely new thing to the mesh generation pipeline.
* We have now got the filters :
* Filters are like post processing items. They are individual effects that can be applied to the mesh.
* There is a erosion filter. You have to complete all base shape generation then disable auto update then apply the filter.

As of now the entire erosion simulation is done on CPU.
Also since the main pipeline is in place one could just modify a single function to add their own erosion algorithm. 

Hope this helps.

**Reply by Jaysmito101** [reply]
> An erosion generator will be nice to have.

Here You GO:   https://youtu.be/wnNclej40zQ

**Comment by razvanab** [comment]
That's amazing nice job and thanks.
I'll try to post something soon. I'm just starting to learn the software.

**Reply by Jaysmito101** [reply]
Thanks!

**Comment by lyn1337** [comment]
I'm seeing some trouble, atleast and perhaps limited to the Linux build, when starting up. The following is thrown on the first startup attempt: 
```
Download Complete - /home/lyn1337/Downloads/TerraForge3D.Linux/Data/cache/github_avatars/lunaboards-dev_MDQ6VXNlcjMwMDg0OTUw
OpenCL Status : Using OpenCL Platform : Clover
OpenCL Status : Using GPU Device : AMD RENOIR (DRM 3.42.0, 5.15.34_1, LLVM 12.0.1)
OpenCL Status : Error Building : fatal error: cannot open file '/usr/share/clc/gfx909-amdgcn-mesa-mesa3d.bc': No such file or directory

```

Please ping if you need a tester!

**Reply by Jaysmito101** [reply]
Hey, thanks for reporting the issue!
I have been getting a lot of issues with OpenCL on Linux.
Also as we have decided to leave opencl and completely move to you compute using OpenGL and Vulkan compute shaders.

Also it would be great to have more testers as currently work is being done to completely rewrite TerraFprge3D from scratch for more platform support, Vulkan support, and many more updates.

You can find more information about that on our community discord server https://discord.gg/mg6VpMMgEK

**Reply by lyn1337** [reply]
Splendid, looking forward to that!

**Comment by Flynsarmy** [comment]
Any chance of a new [release](https://github.com/Jaysmito101/TerraForge3D/releases)? It's been almost exactly 2 years since the last one and I'm sure the app has changed a lot since then.

**Reply by Flynsarmy** [reply]
Your [github.io page](https://jaysmito101.github.io/TerraForge3D/) download button links to v2.2 whereas the newest (currently 2 year old) version is 2.3.

**Reply by Jaysmito101** [reply]
Yes there has been lots of changes basically the app has been rewritten from scratch 2 times(2 diff versions with diff feature sets) to be a whole lot more capable. But no releases yet as it's still being polished and work is left. Plus I don't have as much free time as I had 2 years ago so it's slower development.
